tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/MongoDb/IBsonDocument.cs

```diff
@@ -0,0 +1,24 @@
+// <copyright file="IBsonDocument.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System.ComponentModel;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.MongoDb
+{
+    /// <summary>
+    /// MongoDB.Bson.BsonDocument interface for duck-typing
+    /// </summary>
+    [Browsable(false)]
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public interface IBsonDocument
```

Comment:
We've tended towards interfaces in the past, but @tonyredondo has some benchmarks that show that really we should be using a standard `class` with `virtual` methods instead (where we can't use copy structs), as they are quite a bit faster.

Reply:
Done in 46794d772bf3ede266cdd4202f691014f1a5eb75